server/data/data-jmap/src/main/java/org/apache/james/jmap/api/filtering/Rule.java

```diff
@@ -268,45 +268,40 @@ public String toString() {
 
     public static class Builder {
 
-        private Optional<Id> id;
-        private Optional<String> name;
-        private Optional<Condition> condition;
-        private Optional<Action> action;
+        private Id id;
+        private String name;
+        private Condition condition;
+        private Action action;
 
-        public Builder() {
-            this.id = Optional.empty();
-            this.name = Optional.empty();
-            this.condition = Optional.empty();
-            this.action = Optional.empty();
-        }
+        public Builder() {}
 
         public Builder id(Id id) {
-            this.id = Optional.of(id);
+            this.id = id;
             return this;
         }
 
         public Builder name(String name) {
-            this.name = Optional.of(name);
+            this.name = name;
             return this;
         }
 
         public Builder condition(Condition condition) {
-            this.condition = Optional.of(condition);
+            this.condition = condition;
             return this;
         }
 
         public Builder action(Action action) {
-            this.action = Optional.of(action);
+            this.action = action;
             return this;
         }
 
         public Rule build() {
-            Preconditions.checkState(id.isPresent(), "`id` is mandatory");
-            Preconditions.checkState(name.isPresent(), "`name` is mandatory");
-            Preconditions.checkState(condition.isPresent(), "`condition` is mandatory");
-            Preconditions.checkState(action.isPresent(), "`action` is mandatory");
+            Preconditions.checkState(id != null, "`id` is mandatory");
+            Preconditions.checkState(name != null, "`name` is mandatory");
+            Preconditions.checkState(condition != null, "`condition` is mandatory");
+            Preconditions.checkState(action != null, "`action` is mandatory");
```

Comment:
If they are all required, why not use a staged builder?

Reply:
I moved it in to Rule constructor, forgot this commit (it should be removed).